### PR TITLE
Add context menus

### PR DESCRIPTION
**Pull request by matzipan**
Depends on #102

### CODE REVIEW

**Comment by matzipan** [comment]
This branch's dependency has been unblocked.

File: plugins/folder-manager/FolderItem.vala [hunk 1]
```diff
@@ -22,21 +22,15 @@ namespace Scratch.Plugins.FolderManager {
...
-        //Gtk.MenuItem item_trash;
-        //Gtk.MenuItem item_create;
-
         private GLib.FileMonitor monitor;
         private bool children_loaded = false;
 
-        public FolderItem (File file) requires (file.is_valid_directory) {
-            Object (file: file);
+        public FolderItem (File file, FileView view) requires (file.is_valid_directory) {
+            Object (file: file, view: view);
```
Comment (jeremypw) [inline plugins/folder-manager/FolderItem.vala]: Trailing space.

File: plugins/folder-manager/FolderItem.vala [hunk 2]
```diff
@@ -22,21 +22,15 @@ namespace Scratch.Plugins.FolderManager {
...
         private GLib.FileMonitor monitor;
         private bool children_loaded = false;
 
-        public FolderItem (File file) requires (file.is_valid_directory) {
-            Object (file: file);
+        public FolderItem (File file, FileView view) requires (file.is_valid_directory) {
+            Object (file: file, view: view);        
         }
-
+
```
Comment (jeremypw) [inline plugins/folder-manager/FolderItem.vala]: Trailing space

File: plugins/folder-manager/FolderItem.vala [hunk 3]
```diff
@@ -49,47 +43,71 @@ namespace Scratch.Plugins.FolderManager {
                     children_loaded = true;
                 }
             });
-
+
```
Comment (jeremypw) [inline plugins/folder-manager/FolderItem.vala]: Trailing space

File: plugins/folder-manager/FolderItem.vala [hunk 4]
```diff
@@ -49,47 +43,71 @@ namespace Scratch.Plugins.FolderManager {
...
+            delete_item.activate.connect (() => trash ());
+            menu.append (delete_item);
+            
             menu.show_all ();
             return menu;
-        }*/
-
-        internal void add_children () {
+        }
+
```
Comment (jeremypw) [inline plugins/folder-manager/FolderItem.vala]: Trailing space

**Comment by matzipan** [comment]
I'm closing this branch because of #113 and #114.

**Pull request closed without merging**